src/libraries/System.Private.CoreLib/src/System/IO/Enumeration/FileSystemEntry.Unix.cs

```diff
@@ -32,38 +32,34 @@ namespace System.IO.Enumeration
             entry._pathBuffer = pathBuffer;
             entry._fullPath = ReadOnlySpan<char>.Empty;
             entry._fileName = ReadOnlySpan<char>.Empty;
-
             entry._status.InvalidateCaches();
+            entry._status.InitiallyDirectory = false;
 
             bool isDirectory = directoryEntry.InodeType == Interop.Sys.NodeType.DT_DIR;
             bool isSymlink   = directoryEntry.InodeType == Interop.Sys.NodeType.DT_LNK;
             bool isUnknown   = directoryEntry.InodeType == Interop.Sys.NodeType.DT_UNKNOWN;
 
-            // Some operating systems don't have the inode type in the dirent structure,
-            // so we use DT_UNKNOWN as a sentinel value. As such, check if the dirent is a
-            // symlink or a directory.
-            if (isUnknown)
+            if (isDirectory)
             {
-                isSymlink = entry.IsSymbolicLink;
-                // Need to fail silently in case we are enumerating
-                isDirectory = entry._status.IsDirectory(entry.FullPath, continueOnError: true);
```

Comment:
Tom, do you know which OS sets the `InodeType` to `DT_UNKNOWN`? The tests passed after this was removed, so I'm worried that we weren't testing this code path at all.

The idea of this code was to assume the filesystem entry was a symlink, so we do the additional "is this a directory symlink" check as well.

Reply:
Ah, my bad. This was moved below.
Still, if you have an answer to my initial question, that would be great. I don't know which OS sets this to DT_UNKNOWN.

Reply:
fyi, https://github.com/dotnet/runtime/issues/59947 is about DT_UKNOWN coming from listing files over NFS (in a specific NFS configuration).